frontend/app/routes/_public+/apply+/$id+/date-of-birth.tsx

```diff
@@ -0,0 +1,143 @@
+import { useEffect } from 'react';
+
+import { ActionFunctionArgs, LoaderFunctionArgs, json, redirect } from '@remix-run/node';
+import { Form, useActionData, useLoaderData } from '@remix-run/react';
+
+import { faChevronLeft, faChevronRight } from '@fortawesome/free-solid-svg-icons';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { useTranslation } from 'react-i18next';
+import { z } from 'zod';
+
+import { Button, ButtonLink } from '~/components/buttons';
+import { ErrorSummary, createErrorSummaryItems, hasErrors, scrollAndFocusToErrorSummary } from '~/components/error-summary';
+import { InputField } from '~/components/input-field';
+import { InputLegend } from '~/components/input-legend';
+import { InputSelect } from '~/components/input-select';
+import { getApplyFlow } from '~/routes-flow/apply-flow';
+import { getTypedI18nNamespaces } from '~/utils/locale-utils';
+import { RouteHandleData } from '~/utils/route-utils';
+
+const i18nNamespaces = getTypedI18nNamespaces('eligibility');
+
+export const handle = {
+  breadcrumbs: [
+    // prettier-ignore
+    { labelI18nKey: 'eligibility:breadcrumbs.canada-ca', to: '/personal-information' },
+    { labelI18nKey: 'eligibility:breadcrumbs.benefits' },
+    { labelI18nKey: 'eligibility:breadcrumbs.dental-coverage' },
+    { labelI18nKey: 'eligibility:breadcrumbs.canadian-dental-care-plan' },
+  ],
+  i18nNamespaces,
+  pageIdentifier: 'CDCP-00XX',
+  pageTitleI18nKey: 'eligibility:date-of-birth.page-title',
+} as const satisfies RouteHandleData;
+
+export async function loader({ request, params }: LoaderFunctionArgs) {
+  const applyFlow = getApplyFlow();
+  const { id, state } = await applyFlow.loadState({ request, params });
+
+  return json({ id, state: state.dob });
+}
+
+export async function action({ request, params }: ActionFunctionArgs) {
+  const applyFlow = getApplyFlow();
+  const { id } = await applyFlow.loadState({ request, params });
+
+  const formData = Object.fromEntries(await request.formData());
+  const parsedDataResult = applyFlow.dobSchema.safeParse(formData);
+
+  if (!parsedDataResult.success) {
+    return json({
+      errors: parsedDataResult.error.format(),
+      formData: formData as Partial<z.infer<typeof applyFlow.dobSchema>>,
+    });
+  }
+
+  const sessionResponseInit = await applyFlow.saveState({
+    request,
+    params,
+    state: { dob: parsedDataResult.data },
+  });
+
+  if (getAgeEligibility(parsedDataResult.data.day, parsedDataResult.data.month, parsedDataResult.data.year)) {
+    return redirect(`/apply/${id}/personal-info`, sessionResponseInit);
+  }
+  return redirect(`/apply/${id}/dob-eligibility`, sessionResponseInit);
+}
+
+//Returns true for applicants who are 65 years of age or older
```

Comment:
```suggestion
/**
 * Returns true for applicants who are 65 years of age or older
 */
```

**Reason:** the comment becomes jsdoc and will render when hovering over the function name anywhere in the code.

Reply:
Added in jsdoc (moved to utils folder)